source/fuzz/transformation_move_instruction_down.cpp

```diff
@@ -99,17 +132,26 @@ protobufs::Transformation TransformationMoveInstructionDown::ToMessage() const {
   return result;
 }
 
-bool TransformationMoveInstructionDown::IsOpcodeSupported(SpvOp opcode) {
+bool TransformationMoveInstructionDown::IsInstructionSupported(
+    opt::IRContext* ir_context, const opt::Instruction& inst) {
   // TODO(https://github.com/KhronosGroup/SPIRV-Tools/issues/3605):
   //  We only support "simple" instructions that work don't with memory.
   //  We should extend this so that we support the ones that modify the memory
   //  too.
-  switch (opcode) {
+  return IsSimpleInstruction(ir_context, inst) ||
```

Comment:
I've made this issue more generic since there are possibly many more instructions we could add support for.

Reply:
@Vasniktel should the comment associated with the TODO be updated as well?  Because now we do handle some instructions that work with memory.  The comment looks like this is stale, since the reader of the code can see that there is a whole lot of stuff to do with instructions that do access memory.